File: julia/nop_benchmark.jl
# nop_benchmark.jl
# Julia program to count 'NOPs' (empty statement) executed in 1 second

function nop_benchmark()
    nops = 0
    start = time()
    while time() - start < 1
        # NOP in Julia: empty statement
        nops += 1
    end
    println("NOPs executed: $nops")
end

nop_benchmark()

File: rust/nop_benchmark.rs
// nop_benchmark.rs
// Rust program to count NOPs (empty inline asm) executed in 1 second
use std::time::{SystemTime, Duration};

fn main() {
    let mut nops: u64 = 0;
    let start = SystemTime::now();
    while start.elapsed().unwrap() < Duration::from_secs(1) {
        unsafe { std::arch::asm!("nop"); }
        nops += 1;
    }
    println!("NOPs executed: {}", nops);
}
